client.go

```diff
@@ -0,0 +1,54 @@
+package filetypes
+
+import (
+	"io"
+)
+
+type FileOption func(*Client)
+type Client struct {
+	Writer io.Writer
+	Reader io.Reader
+	// fileFormat     string
+	IncludeHeaders bool
+	Delimiter      rune
+}
+
+func NewClient(options ...FileOption) (*Client, error) {
+	c := &Client{}
+	for _, option := range options {
+		option(c)
+	}
+	c.defaults()
+
+	return c, nil
+}
+
+func WithWriter(w io.Writer) FileOption {
+	return func(c *Client) {
+		c.Writer = w
+	}
+}
+
+func WithReader(r io.Reader) FileOption {
+	return func(c *Client) {
+		c.Reader = r
+	}
+}
+
+func WithHeader(include bool) FileOption {
```

Comment:
This should be just `WithHeader` as by defaults it is `false` so not a lot of value of having a parameter here.